microprofile/WFLY-14987_reactive_messaging-ssl-context-kafka-connector.adoc

```diff
@@ -0,0 +1,278 @@
+
+I looked into forking the Spring utility, but noticed that the system property needed to set up JAAS (`-Djava.security.auth.login.config`) is read at JVM boot. Also, Zookeeper and Kafka need a separate config file for JAAS, and that config file must be specified on JVM startup via a system property. Hence enhancing the embedded Kafka used by the current WildFly tests does not look practical at the moment without writing a lot of process wrappers and starting them as external processes.
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
+////
+The WildFly documentation will be updated to cover this use case.
+
+== Example Setup and Configuration
+For manual testing, I use Zookeeper and Kafka from the download as indicated in the https://kafka.apache.org/quickstart[Apache Kafka Quickstart].
+
+=== Generating the key-/truststores
+To create the certificates:
+Followed  https://docs.actian.com/integrationmanager/2.0/index.html#page/User/Step_1_3a_Create_a_Keystore_File.htm
+
+Create the server keystore with a certificate
+----
+keytool -genkeypair -alias localhost -keyalg RSA -keysize 2048 -storetype PKCS12 -keystore server.keystore.p12 -validity 3650  -ext SAN=DNS:localhost,IP:127.0.0.1
+(Pwd: serverks)
+----
+
+Export the certificate
+----
+keytool -exportcert -alias localhost -keystore server.keystore.p12 -file server.cer -storetype pkcs12 -noprompt -storepass serverks
+----
+Import to the server and client trust stores
+----
+keytool -keystore server.truststore.p12 -alias localhost -importcert -file server.cer -storetype pkcs12
+(Pwd: serverts)
+keytool -keystore client.truststore.p12 -alias localhost -importcert -file server.cer -storetype pkcs12
+(Pwd: clientts)
+----
+=== Configuring Zookeeper and Kafka to work with SASL_SSL
+This is adapted from https://kafka.apache.org/documentation/#security_ssl and https://kafka.apache.org/documentation/#security_sasl, and I use the setup from the https://kafka.apache.org/quickstart[Kafka Quickstart] as the base installation to verify it works.
+
+Following on from the quickstart, I copy the default configuration from the `config` folder of the Apache Kafka distribution to the `config/sasl-ssl` child folder. Next I make the following changes.
+
+*1 - Copy the keystore and truststores in the previous step to the `config/sasl-ssl` folder*
+
+*2 - Add the following files to the `config/sasl-ssl` folder*
+kafka_server_jaas.conf:
+
+[source]
+----
+KafkaServer {
+org.apache.kafka.common.security.plain.PlainLoginModule required
+username="admin"
+password="admin-secret"
+user_admin="admin-secret";
+};
+Client {
+org.apache.zookeeper.server.auth.DigestLoginModule required
```

Comment:
I just want to double check,that mixing DigestLoginModule and PlainLoginModule is what is expected here. Sounds to me strange Plain and Digest should work together, but I definitely do not see deep into it.

Reply:
@mchoma I am not a security expert and it has been so long since I looked into this. However, from my notes I found I was inspired by https://docs.vmware.com/en/VMware-Smart-Assurance/10.1.0/sa-ui-installation-config-guide-10.1.0/GUID-DF659094-60D3-4E1B-8D63-3DE3ED8B0EDF.html so I think it is correct. I think possibly the Plain stuff is for Zookeeper/Kafka communication  and Digest is for clients and Kafka communication.

Reply:
Yes, I assumed it is something like that. Just wanted to make sure.